packages/viewport/src/index.ts

```diff
@@ -89,38 +90,41 @@ function createMediaQueryList( args?: { min?: number; max?: number } ): QueryIte
 	return false;
 }
 
-const mediaQueryLists: Record< string, QueryItem > = {
-	'<480px': createMediaQueryList( { max: 480 } ),
```

Comment:
I think we should create the media query list on demand. Any thoughts?

Reply:
What do you mean on demand?

Reply:
I mean creating them when we need to check the viewport width as they will check the viewport immediately. Hence, we got the result initially in the previous way, but it isn't required.

For example, people might open the page with the small viewport window and then increase the window width. If there is a sub-page (or a component not rendered initially) and need to use the viewport to determine something, the result might be wrong as the initial viewport is small.

Reply:
I see your point 🙂
It seems like a pretty significant change, so it's worth making a simple announcement in dotcom-devs to let people know about this!

Reply:
Posted. See p1686282782913399-slack-C03DY3RJ5